app/models/merchant.rb

```diff
@@ -26,4 +27,15 @@ def distinct_cities
     item_orders.distinct.joins(:order).pluck(:city)
   end
 
+  def distinct_orders
+    orders.distinct
+  end
+
+  def quantity_per_merchant(id)
+    item_orders.where("order_id = ?", id).sum(:quantity)
+  end
+
+  def grandtotal_by_merchant(id)
+    item_orders.where("order_id = ?", id).sum("item_orders.price * quantity")
```

Comment:
I had to add `item_orders` in front of price and then it worked! The "price was ambiguous" error was because the items table also has a price!

Reply:
Great catch!